master/internal/db/postgres_tasks_intg_test.go

```diff
@@ -372,3 +386,312 @@ func TestExhaustiveEnums(t *testing.T) {
+}
+
+func TestAllocationByID(t *testing.T) {
+	tIn := RequireMockTask(t, db, nil)
+	aIn := RequireMockAllocation(t, db, tIn.TaskID)
+
+	a, err := db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn, a)
+}
+
+func TestAllocationSessionFlow(t *testing.T) {
+	uIn := RequireMockUser(t, db)
+	tIn := RequireMockTask(t, db, nil)
+	aIn := RequireMockAllocation(t, db, tIn.TaskID)
+
+	tok, err := db.StartAllocationSession(aIn.AllocationID, &uIn)
+	require.NoError(t, err)
+	require.NotNil(t, tok)
+
+	as, err := allocationSessionByID(t, aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, uIn.ID, *as.OwnerID)
+
+	running := model.AllocationStatePulling
+	aIn.State = &running
+	err = db.UpdateAllocationState(*aIn)
+	require.NoError(t, err)
+
+	a, err := db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn.State, a.State)
+
+	err = db.DeleteAllocationSession(aIn.AllocationID)
+	require.NoError(t, err)
+
+	as, err = allocationSessionByID(t, aIn.AllocationID)
+	require.ErrorContains(t, err, "no rows in result set")
+	require.Nil(t, as)
+}
+
+func TestUpdateAllocation(t *testing.T) {
+	tIn := RequireMockTask(t, db, nil)
+	aIn := RequireMockAllocation(t, db, tIn.TaskID)
+
+	// Testing UpdateAllocation Ports
+	aIn.Ports = map[string]int{"abc": 123, "def": 456}
+	err := UpdateAllocationPorts(*aIn)
+	require.NoError(t, err)
+
+	a, err := db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn.Ports, a.Ports)
+
+	// Testing UpdateAllocationStartTime
+	newStartTime := ptrs.Ptr(time.Now().UTC().Truncate(time.Millisecond))
+	aIn.StartTime = newStartTime
+
+	err = db.UpdateAllocationStartTime(*aIn)
+	require.NoError(t, err)
+
+	a, err = db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn.StartTime, a.StartTime)
+
+	// Testing UpdateAllocationProxyAddress
+	proxyAddr := "here"
+	aIn.ProxyAddress = &proxyAddr
+
+	err = db.UpdateAllocationProxyAddress(*aIn)
+	require.NoError(t, err)
+
+	a, err = db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn.ProxyAddress, a.ProxyAddress)
+}
+
+func TestCloseOpenAllocations(t *testing.T) {
+	// Create test allocations, with a NULL end time.
+	t1In := RequireMockTask(t, db, nil)
+	a1In := RequireMockAllocation(t, db, t1In.TaskID)
+
+	t2In := RequireMockTask(t, db, nil)
+	a2In := RequireMockAllocation(t, db, t2In.TaskID)
+
+	// Set status for both open allocation as 'terminated'.
+	terminated := model.AllocationStateTerminated
+	a1In.State = &terminated
+	a2In.State = &terminated
+
+	// Close only a2In open allocations (filter out the rest).
+	err := db.CloseOpenAllocations([]model.AllocationID{a1In.AllocationID})
+	require.NoError(t, err)
+
+	a1, err := db.AllocationByID(a1In.AllocationID)
+	require.NoError(t, err)
+	require.Nil(t, a1.EndTime)
+
+	a2, err := db.AllocationByID(a2In.AllocationID)
+	require.NoError(t, err)
+	require.NotNil(t, a2.EndTime)
```

Comment:
This test is failing for me, I ran it twice. The code seems correct to me but can you check on your end?

Reply:
huh interesting, as it's passing on circleCI -- I think it's because your db isn't reset between runs, so an allocation that shouldn't exist, does. I'll rewrite the test to be unique on each run.